web-app/app/tricorder/page.tsx

```diff
@@ -1,97 +1,87 @@
 "use client";
 
-import { useEffect, useState } from "react";
+import { useEffect, useRef, useState } from "react";
 import Counter from "./Counter";
-import { Configure } from "./Configure";
+import { Config, ConfigurationDisplay, Configure } from "./Configure";
 
 function Tricorder(props: any) {
-  const scanForLifeforms = useState<any>();
+  const [config, setConfig] = useState<Config>({});
+  const [isScanningForLifeForms, setIsScanningForLifeforms] =
+    useState<boolean>();
+  const [foundLifeforms, setFoundLifeForms] = useState<number[]>([]);
 
-  var isScanningForLifeForms = scanForLifeforms[0];
-  var setScanForLifeFormsToTrue = () => scanForLifeforms[1](true);
-  var setScanForLifeFormsToFalse = () => scanForLifeforms[1](false);
+  const intervalRef = useRef<NodeJS.Timeout>();
 
-  let [foundLifeforms, setFoundLifeForms] = useState<any>([]);
+  const soughtLifeformCount = 47;
 
   useEffect(() => {
-    setInterval(() => {
-      foundLifeforms.push(Math.random());
-    }, 1000);
-  });
+    const multipliers =
+      (config.antimatter ? 0.5 : 1) *
+      (config.slingshot ? 0.5 : 1) *
+      (config.stembolts ? 0.5 : 1);
+    const interval = setInterval(() => {
+      if (isScanningForLifeForms) {
+        setFoundLifeForms((lifeforms) => {
```

Comment:
This is a bit convoluted, since it clears the interval inside the setter. That's probably not the best way to do this! Still, it works for now.